### PR TITLE
feat(_circuit_conversions): add dagger conversions

### PR DESCRIPTION
**Pull request by AthenaCaesura**
## Description

Allow dagger gates in converted circuits.

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [ ] I have updated documentation.

### CODE REVIEW

File: tests/orquestra/integrations/qiskit/conversions/circuit_conversions_test.py [hunk 1]
```diff
@@ -532,14 +532,6 @@ def test_binding_and_exporting_parametrized_circuit_results_in_equivalent_circui
             f"equal to\n{_draw_qiskit_circuit(ref_bound)}"
         )
 
-    def test_converting_circuit_with_daggers_fails_explicitly(self):
```
Comment (mstechly) [inline tests/orquestra/integrations/qiskit/conversions/circuit_conversions_test.py:535]: [issue] You added some new logic which fixes some failure modes we observed. Those failures were not covered by tests, so we learned about them "in the wild". Why you have not added those cases as test cases here?

File: src/orquestra/integrations/qiskit/conversions/_circuit_conversions.py [hunk 2]
```diff
@@ -151,6 +158,26 @@ def _export_gate_via_mapping(gate, applied_qubit_indices, q_register, custom_nam
     return qiskit_cls(*qiskit_params), qiskit_qubits, []
 
 
+def _export_dagger_gate(
```
Comment (mstechly) [inline src/orquestra/integrations/qiskit/conversions/_circuit_conversions.py:161]: [comment] I feel like this whole logic for raising exceptions if export failed got a bit out of hand. Though I don't think it's worth refactoring as a part of this PR, that would be to much hassle.
Could you make a ticket for refactoring it?

**Review by mstechly — CHANGES_REQUESTED** [review]

File: src/orquestra/integrations/qiskit/conversions/_circuit_conversions.py (hunk 2 above)
Comment (AthenaCaesura) [reply src/orquestra/integrations/qiskit/conversions/_circuit_conversions.py:161]: For sure. It's probably slow too.

File: tests/orquestra/integrations/qiskit/conversions/circuit_conversions_test.py (hunk 1 above)
Comment (AthenaCaesura) [reply tests/orquestra/integrations/qiskit/conversions/circuit_conversions_test.py:535]: I'm not sure what made the circuit conversion that was found "in the wild" fail. So I couldn't add a test which checked it. I just know that it worked now. I can use previous versions to find out what the error was. So I suppose I'll do that now and add a test.

**Review by mstechly — APPROVED** [review]

**Pull request merged**